src/halmos/sevm.py

```diff
@@ -1529,6 +1557,9 @@ def run(self, ex0: Exec) -> Tuple[List[Exec], Steps]:
                 elif EVM.ADD <= opcode <= EVM.SMOD: # ADD MUL SUB DIV SDIV MOD SMOD
                     ex.st.push(self.arith(ex, opcode, ex.st.pop(), ex.st.pop()))
 
+                elif EVM.ADDMOD <= opcode <= EVM.MULMOD: # ADDMOD MULMOD
```

Comment:
we always interpret these regardless of `--smt-mod`, right?

Reply:
symbolic mod operations will be uninterpreted, unless `--smt-mod` is provided.

concrete mod operations will be always interpreted, yes.

the following cases are considered as "concrete operations":
- when both operands are concrete
- mod by 0 or 1 (which can be reduced to 0)
- mod by a power of 2 (which can be reduced to lower-bit extraction)
- if `--smt-mod-by-const` is provided

--

the same thing also happens for division operations along with `--smt-div` and `--smt-div-by-const`.

Reply:
see also #103, which replaces the uninterpreted abstraction with the actual symbols for assertion solving in external process.

this way, we can avoid unnecessary nonlinear reasoning for path exploration, and later do full nonlinear reasoning when generating counterexamples.